packages/Search/examples/point_clouds/point_clouds.hpp

```diff
@@ -0,0 +1,213 @@
+        }
+        }
+    }
+}
+
+template <typename Layout, typename DeviceType>
+void filledSphereCloud(
+    double const radius,
+    Kokkos::View<DataTransferKit::Point *, Layout, DeviceType> random_points )
+{
+    static_assert(
+        Kokkos::Impl::MemorySpaceAccess<
+            Kokkos::HostSpace, typename DeviceType::memory_space>::accessible,
+        "The View should be accessible on the Host" );
+    std::default_random_engine generator;
+
+    std::uniform_real_distribution<double> distribution( -radius, radius );
+    auto random = [&distribution, &generator]() {
+        return distribution( generator );
+    };
+
+    unsigned int const n = random_points.extent( 0 );
+    for ( unsigned int i = 0; i < n; ++i )
+    {
+        bool point_accepted = false;
+        while ( !point_accepted )
+        {
+            double const x = random();
+            double const y = random();
+            double const z = random();
+
+            // Only accept points that are in the sphere
+            if ( std::sqrt( x * x + y * y + z * z ) <= radius )
+            {
+                random_points( i ) = {{x, y, z}};
+                point_accepted = true;
+            }
+        }
+    }
+}
+
+template <typename Layout, typename DeviceType>
+void hollowSphereCloud(
+    double const radius,
+    Kokkos::View<DataTransferKit::Point *, Layout, DeviceType> random_points )
+{
+    static_assert(
+        Kokkos::Impl::MemorySpaceAccess<
+            Kokkos::HostSpace, typename DeviceType::memory_space>::accessible,
+        "The View should be accessible on the Host" );
+    std::default_random_engine generator;
+
+    std::uniform_real_distribution<double> distribution( -1., 1. );
+    auto random = [&distribution, &generator]() {
+        return distribution( generator );
+    };
+
+    unsigned int const n = random_points.extent( 0 );
+    for ( unsigned int i = 0; i < n; ++i )
+    {
+        double const x = random();
+        double const y = random();
+        double const z = random();
+        double const norm = std::sqrt( x * x + y * y + z * z );
+
+        random_points( i ) = {
+            {radius * x / norm, radius * y / norm, radius * z / norm}};
+    }
+}
+
+template <typename DeviceType>
+void generatePointCloud(
+    PointCloudType const point_cloud_type, double const length,
+    Kokkos::View<DataTransferKit::Point *, DeviceType> random_points )
+{
+    auto random_points_host = Kokkos::create_mirror_view( random_points );
+    if ( point_cloud_type == PointCloudType::filled_box )
+    {
+        filledBoxCloud( length, random_points_host );
+    }
+    else if ( point_cloud_type == PointCloudType::hollow_box )
+    {
+        hollowBoxCloud( length, random_points_host );
+    }
+    else if ( point_cloud_type == PointCloudType::filled_sphere )
+    {
+        filledSphereCloud( length, random_points_host );
+    }
+    else
+    {
+        hollowSphereCloud( length, random_points_host );
+    }
```

Comment:
I'd prefer `else if ( point_cloud_type == PointCloudType::hollow_sphere ) ... else throw`

Reply:
done

Reply:
Thanks.  You can go ahead and merge when the CI builds are done.